lib/ElementHandle.js

```diff
@@ -111,6 +100,34 @@ class ElementHandle extends JSHandle {
     await this.focus();
     await this._page.keyboard.press(key, options);
   }
+
+  /**
+   * @return {!Promise<Object>}
+   */
+  async boundingBox() {
+    const error = await this.executionContext().evaluate(element => {
+      if (!element.ownerDocument.contains(element))
+        return 'Node is detached from document';
+      if (element.nodeType !== HTMLElement.ELEMENT_NODE)
+        return 'Node is not of type HTMLElement';
+      element.scrollIntoViewIfNeeded();
```

Comment:
I wouldn't expect `element.boundingBox()` to cause side effects like scrolling page. Can we drop it from here?

Reply:
Isn't it the same side effects as keyboard and touchscreen operations (no expecatations there either)?
Maybe it should be optional?
i.e.
```
element.boundingBox({ scrollIntoView: true })
```
(it will return a different bounding box if scroll was needed)

Reply:
> Isn't it the same side effects as keyboard and touchscreen operations (no expecatations there either)?

This only happens for mouse/touch operations. The reason is that we can't click to the off-screen location, it is understandable that element should be on screen to be clicked.

Whereas The `boundingBox` method is a getter, and getters are generally perceived as operations without side-effects.

> (it will return a different bounding box if scroll was needed)

Thanks, I see why you need it. I'd make an internal method `_innerBoundingBox(scrollIntoView)`, that would be used in both `_visibleCenter` and `boundingBox()`:

```js
  async _visibleCenter() {
    const box = this._innerBoundingBox(true /* scrollIntoView */);
    // .. compute center
  }
  async boundingBox() {
    return this._innerBoundingBox(false /* scrollIntoView */);
  }
```

Reply:
I agree about the above, but how do you suggest solve the problem of "off-frame" objects needing to be captured by screenshot.
I would want the behavior of scrolling before calling `boundingBox()`.
`element.scrollIntoViewIfNeeded()` like in `focus()`?